### ISSUE TITLE
Exception in thread "SharedPreferencesImpl-load" java.lang.RuntimeException: java.io.IOException on Windows only

### ISSUE THREAD

**Issue by NeverwinterMoon** [robolectric/robolectric#5186]
### Description
I am getting an exception `Exception in thread "SharedPreferencesImpl-load" java.lang.RuntimeException: java.io.IOException: The filename, directory name, or volume label syntax is incorrect` while running on Windows. The same code works on macOS.

The same problem was already mentioned under https://github.com/robolectric/robolectric/issues/3955 but that issue is closed already.

Stacktrace:
```
Exception in thread "SharedPreferencesImpl-load" java.lang.RuntimeException: java.io.IOException: The filename, directory name, or volume label syntax is incorrect
	at org.robolectric.shadows.OsConstantsValues.canonicalize(OsConstantsValues.java:60)
	at org.robolectric.shadows.OsConstantsValues.getMode(OsConstantsValues.java:50)
	at org.robolectric.shadows.ShadowPosix.stat(ShadowPosix.java:23)
	at libcore.io.Posix.stat(Posix.java)
	at libcore.io.BlockGuardOs.stat(BlockGuardOs.java:299)
	at android.system.Os.stat(Os.java:501)
	at android.app.SharedPreferencesImpl.loadFromDisk(SharedPreferencesImpl.java:114)
	at android.app.SharedPreferencesImpl.access$000(SharedPreferencesImpl.java:52)
	at android.app.SharedPreferencesImpl$1.run(SharedPreferencesImpl.java:90)
Caused by: java.io.IOException: The filename, directory name, or volume label syntax is incorrect
	at java.io.WinNTFileSystem.canonicalize0(Native Method)
	at java.io.WinNTFileSystem.canonicalize(WinNTFileSystem.java:428)
	at java.io.File.getCanonicalPath(File.java:618)
	at org.robolectric.shadows.OsConstantsValues.canonicalize(OsConstantsValues.java:58)
	at org.robolectric.shadows.OsConstantsValues.getMode(OsConstantsValues.java:50)
	at org.robolectric.shadows.ShadowPosix.stat(ShadowPosix.java:23)
	at libcore.io.Posix.stat(Posix.java)
	at libcore.io.BlockGuardOs.$$robo$$libcore_io_BlockGuardOs$stat(BlockGuardOs.java:299)
	at libcore.io.BlockGuardOs.stat(BlockGuardOs.java)
	at android.system.Os.$$robo$$android_system_Os$stat(Os.java:501)
	at android.system.Os.stat(Os.java)
	at android.app.SharedPreferencesImpl.$$robo$$android_app_SharedPreferencesImpl$loadFromDisk(SharedPreferencesImpl.java:114)
	at android.app.SharedPreferencesImpl.loadFromDisk(SharedPreferencesImpl.java)
	at android.app.SharedPreferencesImpl.access$000(SharedPreferencesImpl.java:52)
	at android.app.SharedPreferencesImpl$1.$$robo$$android_app_SharedPreferencesImpl_1$run(SharedPreferencesImpl.java:90)
	at android.app.SharedPreferencesImpl$1.run(SharedPreferencesImpl.java)
```

### Robolectric & Android Version
Tried with Robolectric 4.3 and 4.4-SNAPSHOT and Android SDK 25. Worked on macOS, didn't work on Windows 10.

**Comment by deinlandel** [comment]
Can confirm. Same error on Roboelectirc 4.3. Works on 4.2.1

**Comment by kamgurgul** [comment]
For me this issue appeard after switching to the latests OpenJDK 1.8_242. It is working fine with old official JDK 1.8.0_31.

**Comment by drizzd** [comment]
OsConstantsValues.getMode was introduced in 2d10c50a07a02ae129f94d7d87201ed5e923025a between 4.2.1 and 4.3. In 4.2.1 Os.stat returns 0 for st_mode.

I suppose this is not easy to fix. Either all users of SharedPreferences have to avoid using illegal Windows file names for a preference name. Or Robolectric has to provide a Linux compatible file system for the data directory on Windows.

**Comment by npace** [comment]
+1 to what @deinlandel said - this happens to me on Robolectric 4.3 - 4.4, works fine on 4.2.1

**Comment by hoisie** [comment]
Can someone send a PR that updates ShadowSharedPreferencesImpl on Windows
to convert illegal filename characters into something like hyphens?

**Comment by hoisie** [comment]
This is fixed in HEAD, I'll try to push out a 4.4.1 release that contains this fix.

**Comment by RobbWatershed** [comment]
> This is fixed in HEAD, I'll try to push out a 4.4.1 release that contains this fix.

Any update on that one ? 😁

**Comment by hoisie** [comment]
This is fixed in 4.5-alpha-1, which is available now. If that doesn't work
for you, you can use a custom shadow that duplicates the functionality.

**Comment by ivasku** [comment]
Did anyone solved this ? I have the identical error on windows 11 with Roboelectirc  4.4

**Comment by utzcoz** [comment]
@ivasku The fix(https://github.com/robolectric/robolectric/commit/eb26f9a7d40571ddeea8f950a836d04b16d52533) from @hoisie is released from [robolectric-4.5-alpha-1](https://github.com/robolectric/robolectric/releases/tag/robolectric-4.5-alpha-1). Could you try Robolectric 4.5 or latest Robolectric 4.8.1? The Robolectric 4.8.1 is the best recommendation version to use.
